frontend/src/metabase/containers/ItemPicker/ItemPicker.unit.spec.js

```diff
@@ -294,6 +307,45 @@ describe("ItemPicker", () => {
       expect(items[1]).toHaveTextContent(COLLECTION.REGULAR_2.name);
       expect(items[2]).toHaveTextContent(COLLECTION.REGULAR.name);
     });
+
+    it("should filter collections", async () => {
+      await setup({
+        query: "foo",
+        collectionFilter: (collection, _index, allCollections) =>
+          !isPersonalCollectionOrChild(collection, allCollections) ||
+          collection.id === "root",
```

Comment:
Similar to another comment - does this simplification make sense?

```suggestion
          !isPersonalCollectionOrChild(collection, allCollections)
```